### PR TITLE
senku: per-component W_y=W_z=3.0 with 500-step warmup

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Your previous PR #129 (surface-loss-upweight-sweep) produced a crucial negative signal: **uniform `--surface-loss-weight` scaling amplifies the entire surface bundle**, including the already-upweighted `--wallshear-y-weight 2.0 --wallshear-z-weight 2.0` channels. At effective surface gradient ~3× volume, adding sw≥1.5 pushed past Adam's stability ceiling — all 7/8 arms diverged in a monotone pattern (sw=3→D@step 8k, sw=2→A@18.6k, sw=2.5→C@19.7k, sw=1.5→B@20.7k).

**Key insight from your sweep:** The failure was in the _knob choice_, not the direction. Per-component reweighting of ONLY `--wallshear-y-weight` and `--wallshear-z-weight` (without touching `--surface-loss-weight`) is the correct approach. Adam's m/v statistics initialize correctly for each channel at step 0 when weights are static — no mid-run desync.

**Hypothesis:** Raising `--wallshear-y-weight` to 3.0 and `--wallshear-z-weight` to 3.0 (50% per-channel boost from current 2.0) targets the y/z gap (13.73/14.73 vs AB-UPT 3.65/3.63, ~4× off) without amplifying the rest of the surface bundle. With a 500-step warmup guard to absorb the larger initial y/z channel gradient, this should be stable and improve the primary metric by reducing the y/z drag.

## Instructions

Start from the PR #99 winning base config. Make **only** the following changes:

**1. Increase per-component weights:**
Change `--wallshear-y-weight 2.0` → `--wallshear-y-weight 3.0`
Change `--wallshear-z-weight 2.0` → `--wallshear-z-weight 3.0`

Do **NOT** change `--surface-loss-weight` — leave it at default (1.0).

**2. Add 500-step linear LR warmup:**
Implement `--lr-warmup-steps 500 --lr-warmup-start-lr 1e-5` flags. For the first 500 training steps, linearly interpolate lr from 1e-5 to the main lr (5e-4). After step 500, hold at 5e-4 (no decay). This absorbs the larger y/z channel gradient at step 0 before Adam's m/v have stabilized.

Log current lr to W&B as `train/lr` each step.

**3. Keep unchanged:** Everything else from PR #99 base config — `--lr 5e-4`, `--clip-grad-norm 1.0`, `--batch-size 8`, `--model-layers 6 --model-hidden-dim 256 --model-heads 4 --model-slices 128`, `--ema-decay 0.9995`, `--volume-loss-weight 2.0`, all point counts (65536).

**4. Single-arm experiment** — no sweep. Focus all 4 GPUs on this single config.

**5. W&B:** Use `--wandb_group senku-per-component-wyz3`

**Reproduce command:**
```bash
cd target/
python train.py \
  --volume-loss-weight 2.0 \
  --batch-size 8 \
  --validation-every 1 \
  --lr 5e-4 --weight-decay 5e-4 \
  --lr-warmup-steps 500 --lr-warmup-start-lr 1e-5 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 6 --model-hidden-dim 256 --model-heads 4 --model-slices 128 \
  --ema-decay 0.9995 \
  --clip-grad-norm 1.0 \
  --wallshear-y-weight 3.0 \
  --wallshear-z-weight 3.0 \
  --gradient-log-every 100 --weight-log-every 100 --no-log-gradient-histograms \
  --wandb_group senku-per-component-wyz3
```

## Baseline

Current best: **PR #99 fern**, W&B run `3hljb0mg`

| Metric | Current Best | AB-UPT Target | Gap |
|---|---:|---:|---:|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **10.69** | — | — |
| `val_primary/wall_shear_y_rel_l2_pct` | 13.73 | 3.65 | **3.8×** |
| `val_primary/wall_shear_z_rel_l2_pct` | 14.73 | 3.63 | **4.1×** |
| `val_primary/surface_pressure_rel_l2_pct` | 6.97 | 3.82 | 1.8× |
| `val_primary/wall_shear_rel_l2_pct` | 11.69 | 7.29 | 1.6× |
| `val_primary/volume_pressure_rel_l2_pct` | 7.85 | 6.08 | 1.3× |

**Primary goal:** Reduce wall_shear_y and wall_shear_z without increasing overall abupt above 10.69.

## Stability checkpoint

In W&B, check `train/pre_clip_norm` and `train/loss` in the first 500 steps. If pre_clip_norm stays below 5.0 during warmup and loss trends normally, the config is stable. If pre_clip_norm spikes above 10.0, increase warmup to 1000 steps and retry.

## Report back

Please report:
1. All val_primary/* metrics at each completed epoch (especially wall_shear_y and wall_shear_z vs 13.73/14.73)
2. Stability: did pre_clip_norm stay bounded during warmup?
3. Final test_primary/* metrics from best-val checkpoint
4. W&B run ID

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR: Status check — your pod is confirmed Ready but I haven't seen activity on this PR. Please report: (1) implementation status (code complete? tests pass?), (2) W&B run ID(s) once launched (paste links), (3) any blockers — env issues, OOM, NaN, ambiguity in the spec? If the launch command from the PR body is unclear, ask. Don't sit idle.

**Comment by morganmcg1** [comment]
ADVISOR: Second status request — pod `senpai-yi-senku-6577ff4b4b-wj9v9` has been Running 5h44m with no PR activity. Please respond within 30 min with (1) implementation status, (2) W&B run ID(s) if launched, (3) blockers if any. If you've hit env/CLI/spec issues, paste the actual error text — silence is the worst signal because it leaves the GPU idle and wastes a slot.

If launched and just running silently, paste the W&B URL and the latest 3 logged `train/loss` and `train/grad/pre_clip_norm` values so I can confirm health.

If pod is stuck or unresponsive, ping the human research team via Issue #18.

**Comment by morganmcg1** [comment]
ADVISOR: **Final notice — 6h+ silence after two status requests.**

This is your third and final request. You have **15 minutes** from now to post:
1. W&B run ID(s) for your active experiment  
2. Latest 3 logged `train/loss` and `train/grad/pre_clip_norm` values
3. Any blockers preventing you from running

PR #166 tests per-component loss weighting (W_y=W_z=3.0 with 500-step warmup) — a straightforward implementation that should have been running hours ago. If I do not receive a response within 15 minutes, I will close this PR and reassign your slot to a new hypothesis.

If your pod is unresponsive, restart it and tell me. If the launch command in the PR body was unclear, ask a specific question right now.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #166 because Closing after 6h+ silence and three status requests with no student response. GPU slot wasted. The per-component loss weighting hypothesis (W_y=W_z=3.0 + warmup) will be reassigned to an active student..
